### PR TITLE
Use Oauth2AuthorizedClientProvider to implement authorize clients delegated by azure client.

### PR DESCRIPTION
**Pull request by chenrujun**
Use Oauth2AuthorizedClientProvider to implement authorize clients delegated by azure client.

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @moarychan , @yiliuTo , @zhichengliu12581 , please help to review this PR.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADAuthorizationGrantType.java [hunk 1]
```diff
@@ -5,13 +5,14 @@
...
- * Defines grant types: client_credentials, authorization_code, on_behalf_of.
+ * Defines grant types: client_credentials, authorization_code, on_behalf_of, azure_delegated.
  */
 public enum AADAuthorizationGrantType {
 
     CLIENT_CREDENTIALS("client_credentials"),
     AUTHORIZATION_CODE("authorization_code"),
-    ON_BEHALF_OF("on_behalf_of");
+    ON_BEHALF_OF("on_behalf_of"),
+    AZURE_DELEGATED("azure_delegated");
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADAuthorizationGrantType.java:15]: Related PR: https://github.com/spring-projects/spring-security/pull/10155

**Comment by zhichengliu12581** [comment]
LGTM

**Review by zhichengliu12581 — APPROVED** [review]

**Comment by moarychan** [comment]
LGTM.

**Comment by chenrujun** [comment]
@moarychan , thank you.

**Comment by chenrujun** [comment]
@moarychan , test github notification.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADOAuth2AuthorizedClientRepository.java [hunk 2]
```diff
@@ -1,139 +0,0 @@
...
-        if (result != null) {
-            return (T) result;
-        }
-
-        if (repo.isAzureDelegatedClientRegistration(clientRegistrationId)) {
-            OAuth2AuthorizedClient azureClient = loadAuthorizedClient(getAzureClientId(), principal, request);
-            if (azureClient == null) {
-                throw new ClientAuthorizationRequiredException(AZURE_CLIENT_REGISTRATION_ID);
-            }
-            OAuth2AuthorizedClient fakeAuthzClient = createFakeAuthzClient(azureClient, clientRegistrationId, principal);
```
Comment (querwurzel) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADOAuth2AuthorizedClientRepository.java:79]: For me this looks like an implicit client creation functionality has been removed here .. a "loadOrCreate" so to speak. Other than using the @RegisteredOAuth2AuthorizedClient in controllers how can higher level services now request an authorizedClient which is not the default AZURE client? Thanks in advance

Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADOAuth2AuthorizedClientRepository.java:79]: Hi, @querwurzel.
Thanks for reaching out.
Please refer to `AADAzureDelegatedOAuth2AuthorizedClientProvider` and `AADAuthorizationGrantType.AZURE_DELEGATED` to get more information.

**Pull request merged**